### PR TITLE
Defer BigInt::clone() for new_int

### PR DESCRIPTION
**Pull request by youknowone**
This change will reduce useless `BigInt::clone()` for a few cases.

### CODE REVIEW

File: vm/src/pyobject.rs [hunk 1]
```diff
@@ -384,6 +384,17 @@ impl PyContext {
         PyObject::new(PyInt::new(i), self.int_type(), None)
     }
 
+    #[inline]
+    pub fn new_bigint(&self, i: &BigInt) -> PyObjectRef {
```
Comment (youknowone) [inline vm/src/pyobject.rs]: @coolreader18 Do you have any idea to integrate this function into new_int?

**Comment by coolreader18** [comment]
Sorry for merging #1502 too early, I guess you were still working on it.

**Comment by youknowone** [comment]
That is ok. Actually, i thought it was done too. I am now doing a little refactoring and found related issues.

File: vm/src/pyobject.rs (hunk 1 above)
Comment (coolreader18) [reply vm/src/pyobject.rs]: You could maybe declare your own trait and implement it for both `T where T: Into<BigInt> + ToPrimitive` and `&T where T: Into<BigInt> + ToPrimitive`. I think the real issue is that `ToPrimitive` should also be implemented by `num-traits` for `&T where T: ToPrimitive`.

Comment (youknowone) [reply vm/src/pyobject.rs]: For now, I think I am not interested in working for trait puzzle.
Would you review this PR as is again please?

**Review by coolreader18 — APPROVED** [review]

**Pull request merged**